src/rqt_publisher/publisher.py

```diff
@@ -227,39 +227,59 @@ def _change_publisher_expression(self, publisher_info, topic_name, new_value):
             # Strip topic name from the full topic path
             slot_path = topic_name.replace(publisher_info['topic_name'], '', 1)
             slot_path, slot_array_index = self._extract_array_info(slot_path)
+            slot_type = None
 
-            # Get the property type from the message class
-            slot_type, is_array = \
-                get_slot_type(publisher_info['message_instance'].__class__, slot_path)
```

Comment:
Where did the `get_slot_type()` function call go and why is it no longer necessary? Have you removed the function if it's no longer being used?

Reply:
Added comment to clarify where it's replaced:
```python
# determine the property type, supplemental wrapper for get_slot_type()
slot_type = \
      self._resolve_slot_type(computed_expression, slot_type, slot_path, publisher_info['message_instance'].__class__)
```

Added comments in function to clarify why it's still used:
```python
def _resolve_slot_type(self, expression, slot_type, slot_path, message_class):
        if slot_type is None:
            # check for types that get_slot_type doesn't handle well
            # check for array.array type
            is_array_array = re.search("^array\(\'.\', (\[.*\])\)", expression)
            if is_array_array:
                return array.array

            # check for list type
            is_list = re.match("^\[.*\]\s*$", expression)
            if is_list:
                return list

            # check for string type
            is_string = re.match("^\'.*\'\s*$", expression)
            if is_string:
                return str

            # at this point, get_slot_type can determine the type
            slot_type, _ = \
                get_slot_type(message_class, slot_path)

        return slot_type
```